src/reducers/dialogVisibility.js

```diff
@@ -0,0 +1,25 @@
+import Immutable from 'immutable'
+import { actionReducer } from './utils'
+
+const dialogVisibility = actionReducer(Immutable.fromJS({ showVmDialog: false, showVmDetail: false, showEditTemplate: false }), {
```

Comment:
@mbetak @jelkosz This is basically a hack that I am able to properly handle in sagas.

>  {showVmDialog: true, showVmDetail: true, showEditTemplate: true}
This is actually necessary because the only way to "Open the Doors" is to dispatch showVmDetail as shown bellow, to open addVmDialog:
`function* showAddNewVm (action) {
  yield put(setVmDetailToShow({ vmId: '0' })) // open the gate
 .......... 
 yield put(openVmDialog())
}`
 I know it is not a best solution,  @mbetak has a good suggestion I can try it later.